var/spack/repos/builtin/packages/root/package.py

```diff
@@ -275,179 +275,177 @@ class Root(CMakePackage):
+            define_from_variant('qtgsi', 'qt4'),  # See conflicts
+            define_from_variant('r'),
+            define('rfio', False),
+            define('roofit', False),
+            define_from_variant('root7'),  # See conflicts
+            define('ruby', False),
+            define('sapdb', False),
+            define_from_variant('shadowpw', 'shadow'),
+            define_from_variant('sqlite'),
+            define('srp', False),
+            define_from_variant('ssl'),
+            define_from_variant('table'),
+            define_from_variant('tbb'),
+            define('tcmalloc', False),
+            define_from_variant('tmva'),
+            define_from_variant('unuran'),
+            define_from_variant('vc'),
+            define_from_variant('vdt'),
+            define('veccore', False),
+            define_from_variant('vmc'),
+            define_from_variant('webui', 'root7'),  # requires root7
+            define_from_variant('x11', 'x'),
+            define_from_variant('xft', 'x'),
+            define_from_variant('xml'),
+            define_from_variant('xrootd')
+        ]
 
         # Some special features
         if self.spec.satisfies('@6.20:'):
-            options.append(self.define_from_variant('pyroot', 'python'))
+            options.append(define_from_variant('pyroot', 'python'))
         else:
-            options.append(self.define_from_variant('python'))
+            options.append(define_from_variant('python'))
 
         # #################### Compiler options ####################
 
-        if sys.platform == 'darwin':
-            if self.compiler.cc == 'gcc':
-                options.extend([
-                    '-DCMAKE_C_FLAGS=-D__builtin_unreachable=__builtin_trap',
-                    '-DCMAKE_CXX_FLAGS=-D__builtin_unreachable=__builtin_trap',
-                ])
+        if sys.platform == 'darwin' and self.compiler.cc == 'gcc':
+            cflags = '-D__builtin_unreachable=__builtin_trap'
+            options.extend([
+                define('CMAKE_C_FLAGS', cflags),
+                define('CMAKE_CXX_FLAGS', cflags),
+            ])
 
         # Method for selecting C++ standard depends on ROOT version
-        options.append(('-DCMAKE_CXX_STANDARD={0}' if
-                        self.spec.satisfies('@6.18.00:') else
-                        '-Dcxx{0}:BOOL=ON').format
-                       (self.spec.variants['cxxstd'].value))
+        if self.spec.satisfies('@6.18.00:'):
+            options.append(define_from_variant('CMAKE_CXX_STANDARD', 'cxxstd'))
+        else:
+            options.append(define('cxx' + self.spec.variants['cxxstd'].value,
+                                  True))
 
         if 'mysql-client' in self.spec:
-            options.append('-DCMAKE_PROGRAM_PATH={0}'.format(
-                self.spec['mysql-client'].prefix.bin))
+            options.append(define('CMAKE_PROGRAM_PATH',
+                                  self.spec['mysql-client'].prefix.bin))
```

Comment:
Actually @HadrienG2 this paragraph might not be necessary at all -- it's just adding the mysql-client client's install path to the CMake program search path, but since Spack sets CMAKE_PREFIX_PATH already based on the dependent specs, and that variable is also used to search for executables, I think setting CMAKE_PROGRAM_PATH is redundant.

Reply:
Hi @sethrj, I think it is safe to remove this. ROOT supplies its own `FindMySQL.cmake`, and my reading of this section: 

https://github.com/root-project/root/blob/master/cmake/modules/FindMySQL.cmake#L18-L31 

suggests it'll respect `CMAKE_PREFIX_PATH`.